### PR TITLE
748 - Redo library routes for library versions API

### LINKED ISSUE #748: Improve performance of Docs API server
**Is your feature request related to a problem? Please describe.**
Requests to endpoints like `/api/docs/ids-identity/latest/theme-soho/icons/standard/metadata.json` often take 2s "Waiting (TTFB)", no matter which environment.

**Describe the solution you'd like**
Content loading within 100-200ms.

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #751: "View code on Github" links are not working on Tree component documentation
https://design.infor.com/code/ids-enterprise/latest/tree

### PR DESCRIPTION
**Pull request by clepore**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
The existing routes were taking 2-3s for each call because they were looping through all objects on AWS.

**Related github/jira issue (required)**:
- Fixes #748
- Fixes #751 

**Steps necessary to review your pull request (required)**:
1. Have @nickwynja deploy to staging when ready
    - [ ] Load the code pages and make sure dropdowns/versions all work (including "latest")

### CODE REVIEW

File: src/site/src/app/components/docs-content-page/docs-content-page.component.ts [hunk 1]
```diff
@@ -56,18 +56,13 @@ export class DocsContentPageComponent implements OnInit {
...
-              return versions;
-            })
-            .sort((a, b) => {
-              return semver.compare(a.label, b.label);
-            })
-            .reverse();
+        .subscribe((res: []) => {
+          this.versionPaths = res.map(version => {
+            return {
+              full: `code/${this.library}/${version}`,
```
Comment (nickwynja) [inline src/site/src/app/components/docs-content-page/docs-content-page.component.ts]: I think this needs a leading `/`. Change versions with the dropdown is trying to hit a `/code/code/` path which 404s

Comment (nickwynja) [reply src/site/src/app/components/docs-content-page/docs-content-page.component.ts]:
```suggestion
              full: `${this.library}/${version}`,
```

File: src/site/src/app/components/docs-content-page/docs-content-page.component.ts [hunk 2]
```diff
@@ -56,18 +56,13 @@ export class DocsContentPageComponent implements OnInit {
...
+              label: version
+            };
+          });
 
           this.versionPaths.unshift({
             full: `/${this.library}/latest/`,
```
Comment (nickwynja) [inline src/site/src/app/components/docs-content-page/docs-content-page.component.ts]:
```suggestion
             full: `${this.currentLibrary}/latest/`,
```

File: src/site/src/app/components/sidebar/sidebar.component.ts [hunk 3]
```diff
@@ -69,18 +69,13 @@ export class SidebarComponent implements OnInit, OnChanges {
...
-            return versions;
-          })
-          .sort((a, b) => {
-            return semver.compare(a.label, b.label);
-          })
-          .reverse();
+      .subscribe((res: []) => {
+        this.versionPaths = res.map(v => {
+          return {
+            full: `code/${this.currentLibrary}/${v}`,
```
Comment (nickwynja) [inline src/site/src/app/components/sidebar/sidebar.component.ts]:
```suggestion
            full: `${this.currentLibrary}/${v}`,
```

File: src/site/src/app/components/sidebar/sidebar.component.ts [hunk 4]
```diff
@@ -69,18 +69,13 @@ export class SidebarComponent implements OnInit, OnChanges {
...
+            label: v
+          };
+        });
 
         this.versionPaths.unshift({
           full: `/${this.currentLibrary}/latest/`,
```
Comment (nickwynja) [inline src/site/src/app/components/sidebar/sidebar.component.ts]:
```suggestion
           full: `${this.currentLibrary}/latest/`,
```

**Comment by nickwynja** [comment]
When we roll this out, these are the versions that will need to be added first to get everything working for `latest`:

```
ids-enterprise@4.15.0
  ids-identity@2.2.6
       ids-css@1.4.3
     ids-pendo@1.0.0
```

After that, here are the rest of the versions for each library:

### `ids-enterprise`
- [ ] `4.8.0`
- [ ] `4.9.0`
- [ ] `4.13.0`
- [ ] `4.11.0`
- [ ] `4.14.0`
- [ ] `4.7.0`
- [ ] `4.10.0`
- [ ] `4.15.0`
- [ ] `4.12.1`
- [ ] `4.6.0`
- [ ] `4.6.1`
- [ ] `4.12.0`

### `ids-css`
- [ ] `1.1.0`
- [ ] `1.4.0`
- [ ] `1.0.0`
- [ ] `1.2.3`
- [ ] `1.2.4`
- [ ] `1.3.0`
- [ ] `1.4.3`
- [ ] `1.2.7`
- [ ] `1.2.8`
- [ ] `1.2.0`
- [ ] `1.2.5`
- [ ] `1.4.2`
- [ ] `1.2.2`
- [ ] `1.4.1`

### `ids-identity`

* We only use `latest`, so no other versions need to be added

File: src/site/src/app/components/docs-content-page/docs-content-page.component.ts (hunk 2 above)
Comment (clepore) [reply src/site/src/app/components/docs-content-page/docs-content-page.component.ts]: I'll keep "library" but fix the path

**Pull request merged**